dmsrc/sound_len.dm

```diff
@@ -0,0 +1,33 @@
+// Provided a static RSC file path or a raw text file path, returns the duration of the file in deciseconds as a float.
+/proc/rustg_sound_length(file_path)
+	var/static/list/sound_cache
+	if(isnull(sound_cache))
+		sound_cache = list()
+
+	. = 0
+
+	if(!istext(file_path))
+		if(!isfile(file_path))
+			CRASH("rustg_sound_length error: Passed non-text object")
+
+		if(length("[file_path]"))
+			file_path = "[file_path]"
+		else
+			CRASH("rustg_sound_length does not support non-static file refs.")
+
+	if(!isnull((. = sound_cache[file_path])))
+		return .
```

Comment:
This code is doing too much and is not easy to understand, please write it out more directly

```suggestion
	var/cached_length = sound_cache[file_path]
	if(!isnull(cached_length))
		return cached_length
```